backend/substrapp/tasks/tasks.py

```diff
@@ -797,9 +802,38 @@ def _do_task(client, subtuple_directory, tuple_type, subtuple, compute_plan_id,
     with open(path.join(pred_path, 'perf.json'), 'r') as perf_file:
         perf = json.load(perf_file)
     result['global_perf'] = perf['all']
+
+    tag = subtuple.get("tag")
+    if tag and TAG_VALUE_FOR_TRANSFERT_BUCKET in tag:
+        try:
+            transfer_to_bucket(subtuple["key"], pred_path, model_path)
+        except ClientError as e:
+            logging.error(e)
+
     return result
 
 
+def transfer_to_bucket(tuple_key, pred_path, model_path):
+    access_key = os.getenv('BUCKET_TRANSFERT_ID')
+    secret_key = os.getenv('BUCKET_TRANSFERT_SECRET')
+    bucket_name = os.getenv('BUCKET_TRANSFERT_NAME')
+    if not access_key or not secret_key or not bucket_name:
+        return
+
+    tar_file = f'{tuple_key}.tar.gz'
+    tar_path = path.join('/tmp', tar_file)
+
+    with tarfile.open(tar_path, 'w:gz') as tar:
+        tar.add(pred_path)
+        tar.add(model_path)
+    s3 = boto3.client(
+        's3',
+        aws_access_key_id=access_key,
+        aws_secret_access_key=secret_key)
+    s3.upload_file(tar_path, bucket_name, tar_file)
+    os.remove(tar_path)
```

Comment:
You should do it in a try/except/finally to be sur to remove the file if tar construction or boto raise an exception

Reply:
Indeed, this implementation was a bit optimistic. Fixed